--- a/feeds/packages/net/aria2/files/aria2.init
+++ b/feeds/packages/net/aria2/files/aria2.init
@@ -165,7 +165,8 @@ aria2_start() {
 
 	[ "$enabled" = "1" ] || { _info "Instance \"$section\" disabled."; return 1; }
 	[ -n "$dir" ] || { _err "Please set download dir."; return 1; }
-	[ -d "$dir" ] || { _err "Please create download dir first."; return 1; }
+	mkdir -p "$dir"
+	[ ! -d "$dir" ] && { _err "Failed to create download dir."; return 1; }
 
 	config_file="$config_dir/$NAME.conf.$section"
 	config_file_tmp="$config_dir/$NAME.conf.tmp"
